### PR TITLE
Ref fixes2

### PR DESCRIPTION
**Pull request by kapkic**
Per [lind-project/#221](https://github.com/Lind-Project/lind_project/issues/221).
Several refcount fixes.

### CODE REVIEW

File: src/trusted/service_runtime/nacl_app_thread.c [hunk 1]
```diff
@@ -169,7 +169,8 @@ struct NaClApp *NaClChildNapCtor(struct NaClApp *nap, int child_cage_id, enum Na
     if (!parent_nd) {
       continue;
     }
-
+    
+
```
Comment (rennergade) [inline src/trusted/service_runtime/nacl_app_thread.c]: nitpick but dont need the extra newline here

File: src/trusted/service_runtime/nacl_syscall_common.c [hunk 2]
```diff
@@ -636,15 +635,12 @@ int32_t NaClSysDup2(struct NaClAppThread  *natp,
...
+    NaClDescUnref(new_nd);
   }
   
-  /* We've got to put that old NaClDescriptor back in there... */
-  NaClSetDesc(nap, old_hostfd, old_nd);
-  
+  NaClDescUnref(old_nd);  
   ret = newfd;
-
+
```
Comment (rennergade) [inline src/trusted/service_runtime/nacl_syscall_common.c]: whitespace

**Review by rennergade — CHANGES_REQUESTED** [review]
Just some minor cosmetic changes.

File: src/trusted/service_runtime/nacl_syscall_common.c (hunk 2 above)
Comment (kapkic) [reply src/trusted/service_runtime/nacl_syscall_common.c]: Fixed.

File: src/trusted/service_runtime/nacl_app_thread.c (hunk 1 above)
Comment (kapkic) [reply src/trusted/service_runtime/nacl_app_thread.c]: Fixed.

File: src/trusted/service_runtime/sel_ldr.c [hunk 3]
```diff
@@ -1667,6 +1666,7 @@ void NaClCopyDynamicTextAndVmmap(struct NaClApp *nap_parent, struct NaClApp *nap
             NaClLog(LOG_FATAL, "%s\n", "parent vmmap page NaClMprotect failed!");
           }
         }
+      if(desc) NaClDescUnref(desc);
```
Comment (jesings) [inline src/trusted/service_runtime/sel_ldr.c]: This should be indented one more

**Review by jesings — APPROVED** [review]
Thanks for fixing my wacky fork thing and getting the other stuff working!

File: src/trusted/service_runtime/nacl_app_thread.c [hunk 4]
```diff
@@ -169,7 +169,7 @@ struct NaClApp *NaClChildNapCtor(struct NaClApp *nap, int child_cage_id, enum Na
     if (!parent_nd) {
       continue;
     }
-
+
```
Comment (rennergade) [inline src/trusted/service_runtime/nacl_app_thread.c]: hmm now there is whitespace here...

File: src/trusted/service_runtime/sel_ldr.c (hunk 3 above)
Comment (kapkic) [reply src/trusted/service_runtime/sel_ldr.c]: Shoud be fixed.

File: src/trusted/service_runtime/nacl_app_thread.c (hunk 4 above)
Comment (kapkic) [reply src/trusted/service_runtime/nacl_app_thread.c]: Fixed.

**Review by rennergade — APPROVED** [review]
great!

**Pull request merged**